### PR TITLE
Deprecate `Differentiable.AllDifferentiableVariables`.

### PR DESCRIPTION
**Pull request by dan-zheng**
Remove usages of `AllDifferentiableVariables` and
`var allDifferentiableVariables`.

Friend PR: https://github.com/apple/swift/pull/26527

### CODE REVIEW

**Comment by dan-zheng** [comment]
All references should be removed.
```
$ grep -nr "AllDifferentiableVariables" Sources Tests
# No results.
```

**Comment by Shashi456** [comment]
@dan-zheng there's a mention of `AllDifferentiableVariables" in the readme.

**Comment by dan-zheng** [comment]
> @dan-zheng there's a couple of `AllDifferentiableVariables" in the readme.

I saw that, thanks! I'll fix that in a follow-up PR, since the https://github.com/apple/swift/pull/26527 is sensitive to the commit versions of `tensorflow/swift-apis`.

**Comment by Shashi456** [comment]
@dan-zheng is the loss function refactoring necessary in this PR?

**Comment by dan-zheng** [comment]
> @dan-zheng is the loss function refactoring necessary in this PR?

I'm not sure what exactly you're referring to, but it sounds orthogonal. This patch is meant to be minimal, removing references to `AllDifferentiableVariables`/`allDifferentiableVariables`.

File: Sources/TensorFlow/Loss.swift [hunk 1]
```diff
@@ -287,5 +287,8 @@ public func sigmoidCrossEntropy<Scalar: TensorFlowFloatingPoint>(
 ) -> Tensor<Scalar> {
     // This numerically stable implementation is based on the TensorFlow Python API.
     let maxLogitsWithZero = max(logits, Tensor(0))
-    return reduction(maxLogitsWithZero - logits * labels + log(1 + exp(-abs(logits))))
+    // Note: `result` is split into two lines to avoid the "compiler is unable to type-check this
```
Comment (Shashi456) [inline Sources/TensorFlow/Loss.swift]: @dan-zheng this part.

Comment (dan-zheng) [reply Sources/TensorFlow/Loss.swift]: Yes, it was weirdly necessary to fix compilation, though I'm not sure why. [This comment](https://github.com/apple/swift/pull/26527#issuecomment-518935206) provides a bit more context.

Compilation fixes are necessary 🙂

Comment (Shashi456) [reply Sources/TensorFlow/Loss.swift]: @dan-zheng Ah, Alright thanks for the clarification :)

**Comment by dan-zheng** [comment]
The friend PR https://github.com/apple/swift/pull/26527 has been merged.
`apple/swift` now points to [a commit of `tensorflow/swift-apis` from this PR](https://github.com/apple/swift/blob/2e582b9de229ebb7e9475d8cffeb75bddbdb8649/utils/update_checkout/update-checkout-config.json#L451).

Shall we hold off on merging this PR until the CI toolchain has https://github.com/apple/swift/pull/26527? cc @rxwei

**Comment by rxwei** [comment]
The CI toolchain is already broken unfortunately due to build failures (marcrasi@ is looking into this AFAIK). I think we should merge this.
